src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -319,7 +322,12 @@ def get(self, **kwargs):
         result = self._intersection_of_request_info(request_info)
         if len(result) == 0:
             return []
-        return rows([result])
+        
+        if common_dict == 1:
+            response_list = result.values()
```

Comment:
Well, if there is no identical structure DAS cannot handle it efficiently.
Moreover if structure is unknown it means that we can't write efficient
parser either. This is a big drawback to deal with unstructured data.
As you know searching, parsing of data who change their structure
requires a lot of effort, both CPU and RAM wise.

On  0, ticoann <notifications@github.com> wrote:
> > @@ -319,7 +322,12 @@ def get(self, **kwargs):
> >          result = self._intersection_of_request_info(request_info)
> >          if len(result) == 0:
> >              return []
> > -        return rows([result])
> > +        
> > +        if common_dict == 1:
> > +            response_list = result.values()
> 
> It doesn't have identical structure by nature. i.e. TaskChain has different structure than MC. If you need exact same structure, we first need a survey what information they need which has common for all the request and send only those.  RequestName field is in the value with current api.
> 
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/6700/files#r56181101